src/silo_api/request_handler.test.cpp

```diff
@@ -0,0 +1,74 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "silo_api/manual_poco_mocks.test.h"
+#include "silo_api/request_handler.h"
+
+class MockRequestHandler : public Poco::Net::HTTPRequestHandler {
+  public:
+   MOCK_METHOD(
+      void,
+      handleRequest,
+      (Poco::Net::HTTPServerRequest & request, Poco::Net::HTTPServerResponse& response),
+      ()
+   );
+};
+
+TEST(ErrorRequestHandler, handlesRuntimeErrors) {
+   auto* wrapped_handler_mock = new MockRequestHandler;
+
+   auto under_test = silo_api::ErrorRequestHandler(wrapped_handler_mock);
+
+   ON_CALL(*wrapped_handler_mock, handleRequest)
+      .WillByDefault(testing::Throw(std::runtime_error("my error message")));
+
+   silo_api::test::MockResponse response;
+   silo_api::test::MockRequest request(response);
+   under_test.handleRequest(request, response);
+
+   EXPECT_EQ(response.getStatus(), 500);
```

Comment:
Maybe we could use [pocos responses](https://docs.pocoproject.org/current/Poco.Net.HTTPResponse.html)?